hadoop-hdds/docs/content/feature/Short-Circuit-Read.md

```diff
@@ -0,0 +1,76 @@
+
+This “short-circuit” local read feature will bypass the DataNode, allowing the client to read the file from local disk directly when the client is co-located with the data on the same server.
+
+Short-circuit local read can provide a substantial performance boost to many applications, by removing the overhead of network communication. 
+  
+## Prerequisite
+
+Short-circuit local reads make use of a UNIX domain socket. This is a special path in the filesystem that allows the client and the DataNodes to communicate.
+
+The Hadoop native library "libhadoop.so" provides the support to use the Unix domain socket. Please refer to Native Libraries ("https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-common/NativeLibraries.html") for details of this library.
+
+Before enabling short-circuit local reads, you must have a proper libhadoop.so, and make sure it's under the directory where Java can find and load it through "System.loadLibrary()" call.
+
+The paths that Java will search for libraries are specified by the "java.library.path" property. The default value of "java.library.path" depends on the OS and Java version. For example, on Linux with OpenJDK 8 it is `/usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib`.
+
+Command "ozone checknative" can be used to detect whether libhadoop.so can be loaded successfully by Ozone service. 
+
+## Configuration
+
+Short-circuit local reads need to be configured on both the DataNode and the client. By default, it is disabled.
+
+```XML
+<property>
+   <name>ozone.client.read.short-circuit</name>
+   <value>false</value>
+   <description>Disable or enable the short-circuit local read feature.</description>
+</property>
+```
+
+It makes use of a UNIX domain socket, a special path in the filesystem. You will need to set a path to this socket. 
+
+```XML
+<property>
+   <name>ozone.domain.socket.path</name>
+   <value>/var/lib/ozone_dn_socket</value>
+   <description>The path used to create domain socket.</description>
+</property>
+```
+
+The DataNode needs to be able to create this path. On the other hand, it should not be possible for any user except the Ozone user(user who launches Ozone service) or root to create this path. For this reason, paths under /var/run or /var/lib are often used， just like the current default value "/var/lib/ozone_dn_socket".
+
+If you configure the "ozone.domain.socket.path" to other value, for example "/dir1/dir2/ozone_dn_socket", please make sure that both dir1 and dir2 are exiting directories, and there is no ozone_dn_socket under dir2. ozone_dn_socket will be created by Ozone Datanode later during Datanode start.
+
+### Security Consideration
+
+To ensure data security and integrity, Ozone will follow the rule as Hadoop ("https://cwiki.apache.org/confluence/display/HADOOP2/SocketPathSecurity") on "ozone.domain.socket.path" path permission check. 
```

Comment:
```suggestion
To ensure data security and integrity, Ozone will follow the same rules as Hadoop to check permission on the `ozone.domain.socket.path` path as documented in [Socket Path Security](https://cwiki.apache.org/confluence/display/HADOOP2/SocketPathSecurity). 
```
Is there a better resource for this somewhere on hadoop.apache.org/docs? Linking to confluence instead of official docs is not great, and this link is hardcoded to an old hadoop 2 url that will not get updated with subsequent versions.

Reply:
Also, is our implementation using hadoop linraries to do these permission checks or did we re-implement them ourselves? If we are using hadoop then it makes since to link to hadoop docs so they update with the code. If we re-implemented them ourselves we should also "re-implement" our own documentation so they match now and in the future.

Reply:
The same libhadoop.so is used as Hadoop.  No re-implementation of native code part.